### PR TITLE
Bugfix/gunney/inconsistent allocator distributed closest point

### PR DESCRIPTION
**Pull request by gunney1**
# Fix unneeded copies across memory spaces in `DistributedClosestPoint`

Inconsistent allocator usage placed data in different memory spaces.  This requires copies that cause performance issues in `DistributedClosestPoint`.  We now use the same allocator on all the data used in the search operation.

- This PR is a bugfix
- It does the following:
  - Sets the custom allocator earlier in the set-up to ensure the allocator is used throughout the search.
  - Fixes a couple of places in `DistributedClosestPoint` where the default allocator was used instead of the custom allocator.

### CODE REVIEW

**Review by kennyweiss — APPROVED** [review]
Thanks @gunney1 !

**Review by agcapps — APPROVED** [review]
Thanks, @gunney1

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 1]
```diff
@@ -361,15 +361,20 @@ class DistributedClosestPointImpl
...
 
 public:
-  DistributedClosestPointImpl(RuntimePolicy runtimePolicy, bool isVerbose)
+  DistributedClosestPointImpl(RuntimePolicy runtimePolicy,
+                              int allocatorID,
+                              bool isVerbose)
     : m_runtimePolicy(runtimePolicy)
     , m_isVerbose(isVerbose)
     , m_sqDistanceThreshold(std::numeric_limits<double>::max())
+    , m_allocatorID(axom::INVALID_ALLOCATOR_ID)
```
[the remark this answers is not in the document]
Comment (gunney1) [reply src/axom/quest/DistributedClosestPoint.hpp:370]: Agreed.

File: src/axom/quest/DistributedClosestPoint.hpp [hunk 2]
```diff
@@ -434,6 +409,11 @@ class DistributedClosestPointImpl
     SLIC_ASSERT(allocatorID != axom::INVALID_ALLOCATOR_ID);
     // TODO: If appropriate, how to check for compatibility with runtime policy?
     m_allocatorID = allocatorID;
+    if(m_points.getAllocatorID() != m_allocatorID)
+    {
+      PointArray tmpPoints(m_points, m_allocatorID);
+      m_points.swap(tmpPoints);
+    }
```
[the remark this answers is not in the document]
Comment (gunney1) [reply src/axom/quest/DistributedClosestPoint.hpp:416]: I'll move it to `generateBVHTree`.

**Comment by gunney1** [comment]
Thanks for the prompt reviews @agcapps and @kennyweiss

**Pull request merged**